docs/Integrate GitOps with Secrets Management.md

```diff
@@ -0,0 +1,406 @@
+
+In this documentation we will add volume mounting to a deployment and configure the container pod to use the mounted secret.
+
+*Example deployment file to use the secret provider class*
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  creationTimestamp: null
+  labels:
+    app.kubernetes.io/name: taxi
+    app.kubernetes.io/part-of: app-taxi
+    name: taxi
+    namespace: dev
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+    app.kubernetes.io/name: taxi
+    app.kubernetes.io/part-of: app-taxi
+  strategy: {}
+  template:
+    metadata:
+    creationTimestamp: null
+    labels:
+      app.kubernetes.io/name: taxi
+      app.kubernetes.io/part-of: app-taxi
+    spec:
+      containers:
+        - image: nginxinc/nginx-unprivileged:latest
+          imagePullPolicy: Always
+          name: taxi
+          ports:
+            - containerPort: 8080
+          volumeMounts:
+            - name: secrets-store-inline
+              mountPath: "/mnt/secrets-store"
+              readOnly: true
+          resources: {}
+    serviceAccountName: default
+    volumes:
+      - name: secrets-store-inline
+        csi:
+          driver: secrets-store.csi.k8s.io
+          readOnly: true
+          volumeAttributes:
+          secretProviderClass: "my-aws-provider"
+    status: {}
+```
+Click `REFRESH` on the target application page to apply the updated deployment manifest. Then you can observe that all resources will be successfully synced up.
+
+2. Verification
+List the secrets in the pod mount:
+```
+oc exec <deployment-pod-name> -n dev -- ls /mnt/secrets-store/
+```
+*Example output*
+```
+<your-secret-name>
+```
+View a secret in the pod mount:
+```
+oc exec <deployment-pod-name> -n dev -- cat /mnt/secrets-store/<your-secret-name>
+```
+*Example output*
+```
+<secret_value>
+```
+
+
+## Additional Information
+### Configure your AWS cluster to use AWS Security Token Service (STS)
+Follow [Steps to in-place migrate an OpenShift Cluster to STS](https://github.com/openshift/cloud-credential-operator/blob/master/docs/sts.md#steps-to-in-place-migrate-an-openshift-cluster-to-sts)
+### Obtain the `ccoctl` tool
+It's possible to obtain the ccoctl tool from the [mirror of latest OCP versions](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/) (it's available for the latest versions of all the newer minor releases).
+
+It's also possible to obtain it from the Cloud Credential Operator:
+1. Find the name of the Cloud Credential Operator pod:
+```
+$ oc get pod -n openshift-cloud-credential-operator -l app=cloud-credential-operator
+NAME                                        READY   STATUS    RESTARTS   AGE
+cloud-credential-operator-xxxxxxxxxx-yyyyyy   2/2     Running   0          6h33m
+```
+2. Copy the `ccoctl` binary from the pod to a local directory:
+```
+$ oc cp -c cloud-credential-operator openshift-cloud-credential-operator/<cco_pod_name>:/usr/bin/ccoctl ./ccoctl
```

Comment:
Has someone tried this? I get below error

```
$ oc cp -c cloud-credential-operator openshift-cloud-credential-operator/cloud-credential-operator-5fb569d6bc-hz4gg:/usr/bin/ccoctl ./ccoctl
tar: Removing leading `/' from member names
Dropping out copy after 0 retries
error: unexpected EOF
```

Reply:
@varshab1210 This command worked out for me, can you take another try?
```
➜  ~ oc get pod -n openshift-cloud-credential-operator -l app=cloud-credential-operator
NAME                                         READY   STATUS    RESTARTS   AGE
cloud-credential-operator-6549cbfc7f-tp7zb   2/2     Running   0          58m
➜  ~ oc cp -c cloud-credential-operator openshift-cloud-credential-operator/cloud-credential-operator-6549cbfc7f-tp7zb:/usr/bin/ccoctl ./ccoctl
tar: Removing leading `/' from member names
➜  ~ ls
Applications               application-scc.yaml       hashicorp
Commands                   bin                        iCloud Drive (Archive)
Courses                    ccoctl                     kubernetes_upgrade_script
...
```

Reply:
Probably something with my local set up. It did work for me even though it gave me the above output @svghadi @Srivaralakshmi did you face anything similar?

Reply:
Just FYI, this method is originally from https://access.redhat.com/solutions/7001811

Reply:
@varshab1210 Can you please try this procedure and let me know if this world for you - https://docs.openshift.com/container-platform/latest/installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.html#cco-ccoctl-configuring_configuring-iam-ibm-cloud ?

Reply:
> Probably something with my local set up. It did work for me even though it gave me the above output @svghadi @Srivaralakshmi did you face anything similar?

Nope

Reply:
I have seen such message. You can ignore it. It's not a error.
```
tar: Removing leading `/' from member names
```